### PR TITLE
SoftModeSelector

### PR DESCRIPTION
**Pull request by turian**

### CODE REVIEW

File: torchsynth/module.py [hunk 1]
```diff
@@ -588,3 +588,41 @@ class MonophonicKeyboard(SynthModule):
...
+            ModuleParameterRange(
+                0.0,
+                1.0,
+                name=f"mode{i}weight",
+                description=f"mode{i} weight, before normalization",
+            )
+            for i in range(n_modes)
+        ]
+        super().__init__(synthglobals, **kwargs)
+        self.softmax = torch.nn.Softmax(dim=0)
```
Comment (jorshi) [inline torchsynth/module.py:619]: Don't think we need softmax any more?

File: torchsynth/module.py [hunk 2]
```diff
@@ -588,3 +588,41 @@ class MonophonicKeyboard(SynthModule):
...
+
+    def forward(self) -> Tuple[T, T]:
+        """
+        Normalize all mode weights so they sum to 1.0
+        """
+        # Is this tensor creation slow?
+        # But usually parameter stuff is not the bottleneck
+        params = torch.stack([p.data for p in self.torchparameters.values()])
+        return params / torch.sum(params, dim=0)
```
Comment (jorshi) [inline torchsynth/module.py:628]: Very cool, so idea would be that the mode with the largest value gets selected?

File: torchsynth/module.py (hunk 1 above)
Comment (turian) [reply torchsynth/module.py:619]: REmoving

File: torchsynth/module.py (hunk 2 above)
Comment (turian) [reply torchsynth/module.py:628]: Actually no, right now https://github.com/turian/torchsynth/issues/165

**Review by jorshi — APPROVED** [review]
